web-client/iron-remote-gui/src/services/wasm-bridge.service.ts

```diff
@@ -191,7 +191,7 @@ export class WasmBridgeService {
                                 type: SessionEventType.TERMINATED,
                                 data: 'Session was terminated.',
                             });
-                            return of(err);
+                            throw err;
```

Comment:
I read this code again, and I’m wondering if we need to re-throw the error at all.
```suggestion
```

Reply:
we need, as the signature requires either to return an observable or break the control flow with throw.
`export declare function catchError<T, O extends ObservableInput<any>>(selector: (err: any, caught: Observable<T>) => O): OperatorFunction<T, T | ObservedValueOf<O>>;`

Reply:
alternative is to use `tap` operator instead.

Reply:
we may return a observable with nothing, like the `EMPTY` observable constant. which is practically the same in this case but throw an error really align better with the semantics RxJS provides.

Reply:
Sounds good